sync2jira/upstream_issue.py

```diff
@@ -181,6 +181,126 @@ def handle_github_message(body, config, is_pr=False):
+                ]
+
+                intersection = set(actual) & set(expected)
+                if not intersection:
+                    log.debug(
+                        "None of %r in %r on issue: %s", expected, actual, upstream
+                    )
+                    return None
+            else:
+                # direct comparison
+                actual = body["issue"].get(key)
+                if actual != expected:
+                    log.debug(
+                        "Actual %r %r != expected %r on issue: %s",
+                        key,
+                        actual,
+                        expected,
+                        upstream,
+                    )
+                    return None
+
+    # If this is a dropped issue upstream
+    try:
+        if body["_topic"] == "io.pagure.prod.pagure.issue.drop":
+            body["issue"]["status"] = "Dropped"
+    except KeyError:
+        # Otherwise do nothing
+        pass
+
+    # If this is a tag edit upstream
+    try:
+        # Add all updated tags to the tags on the issue
+        for tag in body["tags"]:
+            body["issue"]["tags"].append(tag)
+    except KeyError:
+        # Otherwise do nothing
+        pass
+
+    # If this is a comment edit
+    try:
+        # Add it to the comments on the issue
+        body["issue"]["comments"].append(body["comment"])
+    except KeyError:
+        # Otherwise do nothing
+        pass
+
+    # Format the assignee field to match github (i.e. in a list)
+    body["issue"]["assignee"] = [body["issue"]["assignee"]]
+
+    return i.Issue.from_pagure(upstream, body["issue"], config)
+
+
+def pagure_issues(upstream, config):
+    """
+    Creates a Generator for all Pagure issues in upstream repo.
+
+    :param String upstream: Upstream Repo
+    :param Dict config: Config Dict
+    :returns: Pagure Issue object generator
+    :rtype: sync2jira.intermediary.Issue
+    """
+    base = config["sync2jira"].get("pagure_url", "https://pagure.io")
+    url = base + "/api/0/" + upstream + "/issues"
+
+    params = config["sync2jira"].get("filters", {}).get("pagure", {}).get(upstream, {})
+
+    response = requests.get(url, params=params)
+    if not bool(response):
+        try:
+            reason = response.json()
+        except Exception:
+            reason = response.text
+        raise IOError("response: %r %r %r" % (response, reason, response.request.url))
+    data = response.json()["issues"]
+
+    # Reformat  the assignee value so that it is enclosed within an array
+    # We do this because Github supports multiple assignees, but JIRA doesn't :(
+    # Hopefully in the future it will support multiple assignees, thus enclosing
+    # the assignees in a list prepares for that support
+    for issue in data:
+        issue["assignee"] = [issue["assignee"]]
+
+    issues = (i.Issue.from_pagure(upstream, issue, config) for issue in data)
+    for issue in issues:
+        yield issue
```

Comment:
Alternatively,
```suggestion
    for issue in data:
        yield i.Issue.from_pagure(upstream, issue, config)
```
which saves creating the extra iterator.